lib/eig_trlm.cpp

```diff
@@ -26,7 +26,8 @@ namespace quda
     beta.resize(n_kr, 0.0);
 
     // Thick restart specific checks
-    if (n_kr < n_ev + 6) errorQuda("n_kr=%d must be greater than n_ev+6=%d\n", n_kr, n_ev + 6);
+    if (n_kr < n_ev + 6) errorQuda("n_kr=%d must be greater than or equal to n_ev+6=%d\n", n_kr, n_ev + 6);
+    if (n_kr < n_conv + 12) errorQuda("n_kr=%d must be greater than or equal to n_conv+12=%d\n", n_kr, n_conv + 12);
```

Comment:
What was the reason for this? Was this a check that was missing all along, or was it empirical?